pkg/services/collector.go

```diff
@@ -0,0 +1,512 @@
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/api/charts", c.ChartmuseumUrl), nil)
+	req.SetBasicAuth(c.ChartmuseumUsername, c.ChartmuseumPassword)
+	resp, err := c.HttpClient.Do(req)
+	if err != nil {
+		panic(err)
+	}
+
+	defer resp.Body.Close()
+	responseBody, err := io.ReadAll(resp.Body)
+	if resp.StatusCode != 200 {
+		log.Fatalf("Receiving list of charts failed. Status code - %d, Body - %s", resp.StatusCode, string(responseBody))
+		os.Exit(1)
+	}
+
+	chartListContainer, err := gabs.ParseJSON(responseBody)
+	if err != nil {
+		panic(err)
+	}
+
+	chartsMap, err := chartListContainer.ChildrenMap()
+	if err != nil {
+		panic(err)
+	}
+
+	for name, _ := range chartsMap {
+		chartsInstances, err := chartListContainer.Path(name).Children()
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+
+		for _, chartInstance := range chartsInstances {
+			chartInstanceMap, err := chartInstance.ChildrenMap()
+			if err != nil {
+				log.Fatal(err)
+				continue
+			}
+
+			chartInstanceVersion := chartInstanceMap["version"].Data().(string)
+			if !slices.Contains(c.ChartVersionsRegistry[name], chartInstanceVersion) {
+				c.ChartVersionsRegistry[name] = append(c.ChartVersionsRegistry[name], chartInstanceVersion)
+			}
+		}
+	}
+}
+
+func (c *Collector) RefreshChartLastRevisionRegistry(secrets *v1.SecretList) {
+	for _, secret := range secrets.Items {
+		if !strings.HasPrefix(secret.Name, "sh.helm.release.v1") {
+			continue
+		}
+		secretChartName, secretRevisionInt, err := getChartNameAndRevisionFromSecretName(secret.Name)
+		if err != nil {
+			log.Fatal(err)
+			continue
+		}
+
+		c.ChartLastRevisionRegistry[secretChartName] = secretRevisionInt
+	}
+}
+
+func (c *Collector) DecodeReleaseString(release string) (*gabs.Container, error) {
+	encodedRelease, err := base64.StdEncoding.DecodeString(release)
+	if err != nil {
+		return nil, err
+	}
+	decodedReleaseReader, err := gzip.NewReader(bytes.NewReader(encodedRelease))
+	if err != nil {
+		return nil, err
+	}
+	decodedReleaseBytes, err := ioutil.ReadAll(decodedReleaseReader)
+	if err != nil {
+		return nil, err
+	}
+
+	return gabs.ParseJSON(decodedReleaseBytes)
+}
+
+func (c *Collector) PackageHelmChart(chartName string, chartVersion string, chartPath string) error {
+	var packageFiles []string
+
+	err := filepath.Walk(chartPath,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			pathIsDirectory, err := isDirectory(path)
+			if err != nil {
+				return err
+			}
+			if !pathIsDirectory {
+				packageFiles = append(packageFiles, path)
+			}
+
+			return nil
+		})
+	if err != nil {
+		return err
+	}
+
+	out, err := os.Create(fmt.Sprintf("%s-%s.tgz", chartName, chartVersion))
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	// Create the archive and write the output to the "out" Writer
+	err = createArchive(packageFiles, fmt.Sprintf("%s/%s/", filepath.Dir(chartPath), filepath.Base(chartPath)), fmt.Sprintf("%s/", chartName), out)
```

Comment:
mb can be replaced with https://github.com/mholt/archiver

Reply:
I decided to use this: https://github.com/helm/helm